src/depth_packet_processor.cpp

```diff
@@ -98,4 +100,64 @@ void DepthPacketProcessor::setFrameListener(libfreenect2::FrameListener *listene
   listener_ = listener;
 }
 
+  DumpDepthPacketProcessor::DumpDepthPacketProcessor() : depth_frame(new Frame(512, 424, 4)), ir_frame(new Frame(512, 424,4)) {
+}
+
+DumpDepthPacketProcessor::~DumpDepthPacketProcessor(){
+  delete depth_frame;
+  delete ir_frame;
+  delete[] p0table_;
+  delete[] xtable_;
+  delete[] ztable_;
+  delete[] lut_;
+}
+
+void DumpDepthPacketProcessor::process(const DepthPacket &packet) {
+    ir_frame->height = 1;
+    ir_frame->width = 1;
+    ir_frame->bytes_per_pixel = packet.buffer_length;
+    ir_frame->timestamp = packet.timestamp;
+    ir_frame->sequence = packet.sequence;
+    ir_frame->data = packet.buffer;
```

Comment:
`packet.buffer` is immediately recycled by the depth stream parser after `process()` ends. `->data` will contain undefined values. You need memcpy here.

My suggestion:
```cpp
...DumpDepthPacketProcessor() : depth_frame(NULL), ...
...
...process(const DepthPacket &packet) {
  depth_frame = new Frame(1, 1, packet.buffer_length); //packet.buffer_length is a fixed value
  depth_frame->timestamp = packet.timestamp;
  depth_frame->sequence = packet.sequence;
  memcpy(depth_frame->data, packet.buffer);
//and make the ir frame a dummy one
  ir_frame = new Frame(1, 1, packet.buffer_length, depth_frame->data);
...
  if (!listener_->onNewFrame(Frame::Depth, depth_frame))
    delete depth_frame; // This way of delete and new can be found in opengl_depth_packet_processor.cpp
  if (!listener_->onNewFrame(Frame::Ir, ir_frame))
    delete ir_frame; // This way of delete and new can be found in opengl_depth_packet_processor.cpp
```